### PR TITLE
[BACK-4387] Update TIDE category boundaries to be inclusive rounded values.

### PR DESCRIPTION
**Pull request by lostlevels**
Updates the TIDE report query predicates to query on rounded percents and be inclusive thresholds instead of exclusive. See comments in code for more details.

### CODE REVIEW

File: patients/repository/repository.go [hunk 1]
```diff
@@ -1517,142 +1517,260 @@ type tideCategory struct {
...
-			{
-				SummaryField:   "timeCGMUsePercent",
-				QueryPredicate: bson.M{"$lte": 0.7},
-			}},
+		Predicates: func(parentFieldPath string) bson.A {
+			field := parentFieldPath + "." + "timeCGMUsePercent"
+			return bson.A{
+				bson.M{field: bson.M{"$lt": 0.70 - halfAPercent}}, // < because round(0.695) = round(69.5%) = 70% so values less than 69.5% round to < 70%.
+				bson.M{"$expr": bson.M{
+					"$lte": bson.A{
```
Comment (ewollesen) [inline patients/repository/repository.go:1630]: This should be $lt.

While no test case for cgm use time was provided in the ticket, it might be worth adding one. Or modifying one of the existing ones to show this distinction.

**Review by ewollesen — CHANGES_REQUESTED** [review]

File: patients/repository/repository.go (hunk 1 above)
Comment (lostlevels) [reply patients/repository/repository.go:1630]: Probably a good idea to add tests around all the boundaries I reckon.

Comment (lostlevels) [reply patients/repository/repository.go:1630]: This is actually correct. The CGM wear time is one of the few categories that uses an exclusive bounds. Added test for it.

Comment (ewollesen) [reply patients/repository/repository.go:1630]: I stand corrected. Do you even need the second rounded criteria for this one? It seems like the `bson.M{field: bson.M{"$lt": 0.70 - halfAPercent}}` should be enough to filter them, no?

**Review by ewollesen — APPROVED** [review]
